authentication/src/main/java/io/camunda/authentication/service/OidcCamundaUserService.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.service;
+
+import io.camunda.authentication.entity.AuthenticationContext;
+import io.camunda.authentication.entity.CamundaOidcUser;
+import io.camunda.authentication.entity.CamundaUserDTO;
+import io.camunda.search.entities.RoleEntity;
+import io.camunda.security.entity.ClusterMetadata.AppName;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Profile("auth-oidc")
+@Service
+public class OidcCamundaUserService implements CamundaUserService {
+  private static final String SALES_PLAN_TYPE = "";
+  private static final Map<AppName, String> C8_LINKS = Map.of();
+
+  private Optional<CamundaOidcUser> getCamundaUser() {
+    return Optional.ofNullable(SecurityContextHolder.getContext().getAuthentication())
+        .map(Authentication::getPrincipal)
+        .map(principal -> principal instanceof final CamundaOidcUser user ? user : null);
+  }
+
+  @Override
+  public CamundaUserDTO getCurrentUser() {
+    return getCamundaUser()
+        .map(
+            user -> {
+              final AuthenticationContext auth = user.getAuthenticationContext();
+              return new CamundaUserDTO(
+                  user.getUserInfo().getSubject(),
+                  null,
+                  user.getDisplayName(),
+                  user.getName(),
+                  user.getEmail(),
+                  auth.authorizedApplications(),
+                  auth.tenants(),
+                  auth.groups(),
+                  auth.roles().stream().map(RoleEntity::name).toList(),
+                  SALES_PLAN_TYPE,
+                  C8_LINKS,
+                  true);
+            })
+        .orElse(null);
+  }
+
+  @Override
+  public String getUserToken() {
+    return "";
```

Comment:
💭 Does this need to be an empty string or can we thrown an exception instead?

Reply:
I believe this method is unused. You added it in https://github.com/camunda/camunda/pull/24947/files#diff-cb31ab0a8e3d005cede786bc73cb750cad552a4e4096c9d894592b2eac80b22dR15.
I thought we would eventually need it to implement UserReader.getUserToken(), but that's only used for /v1/internal/users/token and maybe that can be removed too.

Reply:
Ah yes, this was for backwards compatibility. I think we can leave it as this for now